### PR TITLE
Reducing CRs in Object Identification

### PR DESCRIPTION
**Pull request by profxj**
Primary advance here is to reduce the effect of CRs on
object identification.  This is performed by sigma_clipping
the rectified image in artrace.trace_object() prior to smashing
it into trcprof.

The change may make it less likely to detect a bright emission
line source, but those were going to be challenging anyhow.
We can make the sigma_clipping a user-option instead.

Also removes objects identified within 3% of the slit edge, by default.

Tested extensively on LRISr data.  False positives are reduced although
not entirely eliminated.

New docs
No new tests

### CODE REVIEW

**Comment by profxj** [comment]
Addresses Issue #177

File: doc/object_id.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+Overview
+========
+
+Object identification is a challenging process to
+code, especially to allow for a large dynamic range
+between bright continuum sources and faint emission
+line sources.   Our general philosophy has been to
+err on the faint side, i.e. aggressively
+detect sources aggressively with the side-effect of
```
Comment (rcooke-ast) [inline doc/object_id.rst]: one too many aggressively  :-)

File: doc/object_id.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+The standard algorithm performs the following steps:
+
+1. Rectify the sky-subtracted frame
+
+2. Smooth this 2D image
+
+3. Perform sigma clipping (median stat) down the wavelength dimension to further reject CRs.  This may eliminate bright emission lines.
+
+4.  Smash the 2D image into a 1D array aligned with the slit (i.e. spatial dimension).
```
Comment (rcooke-ast) [inline doc/object_id.rst]: Could be clearer. I think you mean (or something along these lines):

"Smash the 2D image along the spectral dimension, to get a 1D array that represents the spatial profile of the exposure."

**Comment by rcooke-ast** [comment]
Looks good - merge when ready.

File: doc/object_id.rst (hunk 2 above)
Comment (profxj) [inline doc/object_id.rst]: Agreed. Edited

**Comment by profxj** [comment]
Merging

**Pull request merged**